domain/src/main/java/org/oppia/domain/analytics/AnalyticsController.kt

```diff
@@ -0,0 +1,184 @@
+package org.oppia.domain.analytics
+
+import android.content.Context
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLog.EventAction
+import org.oppia.app.model.EventLog.Priority
+import org.oppia.util.logging.EventLogger
+import javax.inject.Inject
+
+const val TEST_TIMESTAMP = 1556094120000
+const val TEST_TOPIC_ID = "test_topicId"
+const val TEST_STORY_ID = "test_storyId"
+const val TEST_EXPLORATION_ID = "test_explorationId"
+const val TEST_QUESTION_ID = "test_questionId"
+
+/**
+ * Controller for handling analytics event logging.
+ * [logTransitionEvent] should be used to log screen transition events.
+ * [logClickEvent] should be used to log button click events.
+ */
+class AnalyticsController @Inject constructor(
+  private val eventLogger: EventLogger
+) {
+  /**
+   * Logs transition events.
+   * These events are given HIGH priority.
+   */
+  fun logTransitionEvent(
+    context: Context,
+    timestamp: Long,
+    eventAction: EventAction,
+    explorationContext: EventLog.ExplorationContext?,
+    questionContext: EventLog.QuestionContext?,
+    topicContext: EventLog.TopicContext?,
+    storyContext: EventLog.StoryContext?
```

Comment:
Instead of creating 4 separate fields, lets just pass one field of type context. Callers can create the appropriate context and pass it here.

Reply:
Okay, will do.

Reply:
Done.